include/mbedtls/ecp.h

```diff
@@ -227,391 +250,435 @@ mbedtls_ecp_keypair;
+ *                  for that.
  *
- * \return          0 if successful,
- *                  MBEDTLS_ERR_ECP_BAD_INPUT_DATA if input is invalid,
- *                  MBEDTLS_ERR_MPI_ALLOC_FAILED if memory allocation failed,
- *                  MBEDTLS_ERR_ECP_FEATURE_UNAVAILABLE if the point format
+ * \param grp       The group to which the point should belong.
+ * \param P         The point to import.
+ * \param buf       The input buffer.
+ * \param ilen      The length of the input.
+ *
+ * \return          \c 0 on success.
+ * \return          #MBEDTLS_ERR_ECP_BAD_INPUT_DATA if input is invalid.
+ * \return          #MBEDTLS_ERR_MPI_ALLOC_FAILED on memory-allocation failure.
+ * \return          #MBEDTLS_ERR_ECP_FEATURE_UNAVAILABLE if the point format
  *                  is not implemented.
  *
- * \note            This function does NOT check that the point actually
- *                  belongs to the given group, see mbedtls_ecp_check_pubkey() for
- *                  that.
  */
 int mbedtls_ecp_point_read_binary( const mbedtls_ecp_group *grp, mbedtls_ecp_point *P,
                            const unsigned char *buf, size_t ilen );
 
 /**
- * \brief           Import a point from a TLS ECPoint record
+ * \brief           This function imports a point from a TLS ECPoint record.
  *
- * \param grp       ECP group used
- * \param pt        Destination point
- * \param buf       $(Start of input buffer)
- * \param len       Buffer length
+ * \note            On function return, \p buf is updated to point to immediately
+ *                  after the ECPoint record.
  *
- * \note            buf is updated to point right after the ECPoint on exit
+ * \param grp       The ECP group used.
+ * \param pt        The destination point.
+ * \param buf       The address of the pointer to the start of the input buffer.
+ * \param len       The length of the buffer.
  *
- * \return          0 if successful,
- *                  MBEDTLS_ERR_MPI_XXX if initialization failed
- *                  MBEDTLS_ERR_ECP_BAD_INPUT_DATA if input is invalid
+ * \return          \c 0 on success.
+ * \return          An \c MBEDTLS_ERR_MPI_XXX error code on initialization failure.
+ * \return          #MBEDTLS_ERR_ECP_BAD_INPUT_DATA if input is invalid.
  */
 int mbedtls_ecp_tls_read_point( const mbedtls_ecp_group *grp, mbedtls_ecp_point *pt,
                         const unsigned char **buf, size_t len );
 
 /**
- * \brief           Export a point as a TLS ECPoint record
+ * \brief           This function exports a point as a TLS ECPoint record.
  *
- * \param grp       ECP group used
- * \param pt        Point to export
- * \param format    Export format
- * \param olen      length of data written
- * \param buf       Buffer to write to
- * \param blen      Buffer length
+ * \param grp       The ECP group used.
+ * \param pt        he point format to export to an \c MBEDTLS_ECP_PF_XXX constant.
```

Comment:
Typo: `he` -> `The`

Reply:
I think a comma is missing after `to`, i.e. `The point format to export to, an \c MBEDTLS_ECP_PF_XXX constant.`

Reply:
fixed "The".
Re. comma - this changes the meaning understood by the sentence - can you clarify what "an \c MBEDTLS_ECP_PF_XXX constant" means here?

Reply:
An `MBEDTLS_ECP_PF_XXX` constant is how one specifies a point format. It is not the destination of the export, so it can't follow “export to” with no punctuation. Hence: “the point format to export to, an `\c MBEDTLS_ECP_PF_XXX` constant” (a rather typical way of expressing this kind of thing) or “the point format to export to (an `\c MBEDTLS_ECP_PF_XXX` constant)” or (but I find this unsuitably long) “the point format to export to, which should be expressed as an `\c MBEDTLS_ECP_PF_XXX` constant” or some variation thereof.

Reply:
What about the following:
"the \c MBEDTLS_ECP_PF_XXX point format to export to"?
which follows the "<type> <object>" structure. For example: "white fence" or "Mbed TLS api" or "AES cipher"

Reply:
This does have all the elements, but their arrangement is bizarre. `MBEDTLS_ECP_PF_XXX` is not a specific kind of point format (which would typically be expressed as an adjective or as a noun in adjective position) but an alternative way to state “the point format” (which is naturally expressed as an apposition).

Reply:
So, the closest I can think of is, instead of:
The point format to export to (an \c MBEDTLS_ECP_PF_XXX constant)

The more accurate English phrasing would be:
The point format (an \c MBEDTLS_ECP_PF_XXX constant) to export to.

As "(an \c MBEDTLS_ECP_PF_XXX constant)" refers to the point format.
Is this acceptable?

Reply:
That's unusual but acceptable. I find it clearer not to split “the point format to export to” with a parenthetical statement, however. The two phrase orders have the same meaning, and I don't see what stylistic rule could lead to preferring splitting.

Reply:
The original phrasing, with comma added, does not clearly convey what you say it means:
"The point format to export to, an \c MBEDTLS_ECP_PF_XXX constant."
If you do not know the product intimately (==customers), the connection between the two parts is unclear.
How about:
"The point format to export to. The point format is an \c MBEDTLS_ECP_PF_XXX constant."

Reply:
done